src/directives/intersect/index.ts

```diff
@@ -22,20 +24,37 @@ const setUpObserver: DirectiveHook = (
             initiated: false,
             observer: new IntersectionObserver((entries) => {
                 const entry = entries.find(e => e.isIntersecting);
+                const modifiers = binding.modifiers as Modifiers;
 
-                binding.modifiers.idle
+                const cb = modifiers.idle
                     // eslint-disable-next-line @typescript-eslint/no-unsafe-call
-                    ? requestIdleCallback(() => (binding.value as IntersectCallback)(!!entry, entry))
-                    : (binding.value as IntersectCallback)(!!entry, entry);
+                    ? () => requestIdleCallback(() => (binding.value as IntersectCallback)(!!entry, entry))
+                    : () => (binding.value as IntersectCallback)(!!entry, entry);
+
                 // callback runs on setup so push to the next event cycle
                 initiate(el);
 
-                if (binding.modifiers.once && el._intersectionObserver?.initiated) {
-                    unbindObserver(el);
+                if (el._intersectionObserver?.initiated) {
```

Comment:
Seems to be a bit of duplication of lines 29-52 and 71-94. Have you given slimming this down a go?

Reply:
Unfortunately it isn't a duplication, close... but idk hot wo make it more consise.